packages/components/src/theme/useMonacoTheme.tsx

```diff
@@ -0,0 +1,149 @@
+// react functional component
+
+import React, { useState, useRef, useEffect, useCallback } from 'react';
+import * as monaco from 'monaco-editor';
+import { ThemeContextType } from './ThemeProvider';
+
+/**
+ * Function to walk an object of key values, nested objects, and objects
+ * nested in arrays, and then update the values in a new object.
+ * @example
+ * const obj = {
+ *  rules: [{
+ *   token: 'keyword',
+ *   foreground: '--spectrum-blue-visual-color',
+ *   }],
+ * colors: {
+ *  'editor.background': '--spectrum-gray-50'
+ *  };
+ * const newColors = walkObject(obj, removeHashtag);
+ * console.log(newColors);
+ */
+function walkObject(
+  obj: { [key: string]: unknown },
+  fn: (v: unknown) => unknown
+) {
+  if (obj == null) return obj;
+  const newObj = obj;
```

Comment:
If this is intended to create a new object, cloning should probably happen here instead of outside.